src/main/java/org/pkcs11/jacknji11/AttributeLengthStrategy.java

```diff
@@ -40,6 +40,14 @@ public interface AttributeLengthStrategy {
      * and their maximum length as well as a default length for regular attributes.
      * <p>
      * Use to avoid querying length of attributes for every call to C_GetAttributeValue.
+     * <p>
+     * Note that implementations need to adhere strictly to the PKCS#11 for this strategy 
+     * to work. One detail where implementations (even popular ones) are known to be
+     * non-compliant is returning CKR_BUFFER_TOO_SMALL and setting uLVlaueLength to 
+     * CK_UNAVAILABLE_INFORMATION if sent a pre-allocated but too small buffer.
```

Comment:
Actually that is correct behavior.

Reply:
Yes indeed, it's a strange reverse logic formulation from my side. I'll improve on it.

Reply:
Added PR #60 with improved javadoc. Also modified some of the other javadoc as 8K RSA keys are fairly common in production, but definitely typically not in performance critical applications.